src/System.Security.Cryptography.X509Certificates/tests/CollectionTests.cs

```diff
@@ -0,0 +1,379 @@
+                }
+            }
+        }
+
+        [Fact]
+        public static void ExportCert()
+        {
+            TestExportSingleCert(X509ContentType.Cert);
+        }
+
+        [Fact]
+        public static void ExportSerializedCert()
+        {
+            TestExportSingleCert(X509ContentType.SerializedCert);
+        }
+
+        [Fact]
+        public static void ExportSerializedStore()
+        {
+            TestExportStore(X509ContentType.SerializedStore);
+        }
+
+        [Fact]
+        public static void ExportPkcs7()
+        {
+            TestExportStore(X509ContentType.Pkcs7);
+        }
+
+        private static void TestExportSingleCert(X509ContentType ct)
+        {
+            using (var msCer = new X509Certificate2(TestData.MsCertificate))
+            using (var pfxCer = new X509Certificate2(TestData.PfxData, TestData.PfxDataPassword))
+            {
+                X509Certificate2Collection cc = new X509Certificate2Collection(new X509Certificate2[] { msCer, pfxCer });
+
+                byte[] blob = cc.Export(ct);
+
+                Assert.Equal(ct, X509Certificate2.GetCertContentType(blob));
+
+                X509Certificate2Collection cc2 = new X509Certificate2Collection();
+                cc2.Import(blob);
+                int count = cc2.Count;
+                Assert.Equal(1, count);
+
+                using (X509Certificate2 c = cc2[0])
+                {
+                    Assert.NotSame(msCer, c);
+                    Assert.NotSame(pfxCer, c);
+
+                    Assert.True(msCer.Equals(c) || pfxCer.Equals(c));
+                }
+            }
+        }
+
+        private static void TestExportStore(X509ContentType ct)
+        {
+            using (var msCer = new X509Certificate2(TestData.MsCertificate))
+            using (var pfxCer = new X509Certificate2(TestData.PfxData, TestData.PfxDataPassword))
+            {
+                X509Certificate2Collection cc = new X509Certificate2Collection(new X509Certificate2[] { msCer, pfxCer });
+
+                byte[] blob = cc.Export(ct);
+
+                Assert.Equal(ct, X509Certificate2.GetCertContentType(blob));
+
+                X509Certificate2Collection cc2 = new X509Certificate2Collection();
+                cc2.Import(blob);
+                int count = cc2.Count;
+                Assert.Equal(2, count);
+
+                X509Certificate2[] cs = cc2.ToArray().OrderBy(c => c.Subject).ToArray();
+
+                using (X509Certificate2 first = cs[0])
+                {
+                    Assert.NotSame(msCer, first);
+                    Assert.Equal(msCer, first);
+                }
+
+                using (X509Certificate2 second = cs[1])
+                {
+                    Assert.NotSame(pfxCer, second);
+                    Assert.Equal(pfxCer, second);
+                }
+            }
+        }
+
+        private static X509Certificate2[] ToArray(this X509Certificate2Collection col)
```

Comment:
This one, I think, is because System.Linq doesn't like .NET 1.0/1.1 typed collections.  One would have to run it through coll.Cast<X509Certificate2> (or OfType) in order to call ToArray() and have the strong reference.

Unless I'm entirely mistaken, of course.